ocaml/xapi/xapi_vm.ml

```diff
@@ -238,6 +238,35 @@ let assert_memory_constraints ~__context ~vm platformdata =
    same VM twice during memory calculations to determine whether a given VM can start on a particular host..
 *)
 
+let update_platform_secureboot ~__context ~self platform =
+  if List.assoc "secureboot" platform = "auto" then
+    let t = ("secureboot", string_of_bool (Db.Pool.get_uefi_certificates ~__context ~self <> "")) in
+    List.append [t] (List.remove_assoc "secureboot" platform)
+  else
+    platform
+
+let save_uefi_certificates_to_dir ~__context ~self =
+  let dir = ref [||] in
```

Comment:
Why is this a reference? I expected something like
```
let dir = try Sys.readdir !Xapi_globs.varstore_dir with Sys_error -> ..; [| (*empty *) |] in
```
as there doesn't seem to be an update to `dir`.

Reply:
Thank you. I struggled to get the scoping of the exception handling to work for me with the dir being accessed afterwards